lua/gitlinker/git.lua

```diff
@@ -127,7 +131,7 @@ end
 
 local function parse_port(stripped_uri, host)
   assert(host)
-  local port_capture = host .. ":([0-9]+)[:/].+$"
+  local port_capture = hyphens_escaped(host) .. ":([0-9]+)[:/].+$"
```

Comment:
I think the proper solution is stop using `host` (or `hyphens_escaped(host)`) in regex captures. It's dangerous because it can contain other magic chars.

I would prefer something like

```suggestion
  local port_capture =  chars .. "+:([0-9]+)[:/].+$"
```